commercetools-sdk-base/src/main/java/io/sphere/sdk/utils/CompletableFutureUtils.java

```diff
@@ -64,49 +124,178 @@ private CompletableFutureUtils() {
         });
     }
 
+    /**
+     * Creates a {@link CompletionStage} which may recovered from an occurred error. Alias for {@link CompletionStage#exceptionally(Function)}.
+     * If the recovery function also requires to create a {@link CompletionStage} then use {@link #recoverWith(CompletionStage, Function)}.
+     *
+     * {@include.example io.sphere.sdk.utils.CompletableFutureUtilsTest#recoverFailure()}
+     *
+     * @param future future which may completes exceptionally
+     * @param f function how the exception should be handled to create a successfully completed future, but it also can throw exceptions
+     * @param <T> type of the value of the future
+     * @return a future which may recovered from errors
+     */
     public static <T> CompletionStage<T> recover(final CompletionStage<T> future, final Function<Throwable, ? extends T> f) {
         return future.exceptionally(f);
     }
 
-    public static <T> CompletionStage<T> recoverWith(final CompletionStage<T> future, final Function<? super Throwable, CompletionStage<T>> f) {
+    /**
+     * Creates a {@link CompletionStage} which may recovered from an occurred error.
```

Comment:
Actually I would rather say "which is recovered from an error, in case an error occurs", since the possibility is on the error occurring, not on the recoverability. Or simply "which is recovered if an error occurs".